lib/Dialect/FIRRTL/Transforms/LowerClasses.cpp

```diff
@@ -496,10 +499,29 @@ PathTracker::processPathTrackers(const AnnoTarget &target) {
       } else {
         moduleName = cast<hw::InnerRefAttr>(oldPath.front()).getModule();
       }
+
+      // Copy the leading part of the hierarchical path from the owning module
+      // to the start of the annotation's NLA.
+      InstanceGraphNode *node = instanceGraph.lookup(moduleName);
+      while (true) {
+        // If we get to the owning module or the top, we're done.
+        if (node->getModule() == owningModule || node->noUses())
+          break;
+
+        // Append the next level of hierarchy to the path. Note that if there
+        // are multiple instances, which we warn about, this is where the
+        // ambiguity manifests. In practice, just picking usesBegin generates
+        // the same output as EmitOMIR would for now.
+        InstanceRecord *inst = *node->usesBegin();
+        path.push_back(
+            OpAnnoTarget(inst->getInstance<InstanceOp>())
+                .getNLAReference(namespaces[inst->getParent()->getModule()]));
```

Comment:
We need to use the inner symbol namespace here from the current instances parent, so we can't use `moduleNamespace`, since that's not the right namespace. This is why I added the full inner symbol namespace collection as an instance variable. I think this should be safe, but wanted to call it out.